src/awesome/class.rs

```diff
@@ -0,0 +1,163 @@
+//! Utility methods and constructors for Lua classes
+
+use std::default::Default;
+use std::rc::Rc;
+use rlua::{self, Lua, ToLua, Table, UserData, AnyUserData, Value, Function};
+use super::object::Object;
+use super::property::Property;
+
+pub type Allocator = Rc<Fn(&Lua) -> rlua::Result<Object>>;
+pub type Collector = Rc<Fn(Object)>;
+pub type PropF<'lua> = rlua::Function<'lua>;
+pub type Checker = Rc<Fn(Object) -> bool>;
+
+#[derive(Debug)]
+pub struct Class<'lua> {
+    table: Table<'lua>
+}
+
+#[derive(Clone)]
+pub struct ClassState {
+    name: String,
+    // TODO Needed? If we store this like the object surely it's not needed...
+    /// The global Lua key that corresponds to this class' signal list.
+    /// Stored this way so that it's not tied to Lua lifetime.
+    signals_global: Option<String>,
```

Comment:
Move to meta table directly and eschue the weird stringly type? That would make it actually garbage collector counted too!